gravity.sh

```diff
@@ -82,6 +87,21 @@ if [[ -r "${piholeDir}/pihole.conf" ]]; then
   echo -e "  ${COL_LIGHT_RED}Ignoring overrides specified within pihole.conf! ${COL_NC}"
 fi
 
+# Create pihole diagnosis message
+create_diagnosis_message() {
+  declare result=""
+  # Only create the message if running non-interactive (cron)
+  if pstree -apl "$(pgrep -n cron)" | grep -q "$(basename "$0")"; then
+    # prevent duplicates, change the timestamp.
+    result=$(pihole-FTL sqlite3  "${FTLDBfile}" "SELECT * FROM message WHERE message LIKE '%$1%';")
+    if [ -z "${result}" ]; then
+      sudo pihole-FTL sqlite3 "${FTLDBfile}" "INSERT OR IGNORE into message ( timestamp, type, message, blob1 ) values ('$(date +%s)', 'SCRIPT', '$1', '$(base64 <<< "$(basename "$0")")');"
+    else
+      sudo pihole-FTL sqlite3 "${FTLDBfile}" "UPDATE message SET timestamp = '$(date +%s)';"
```

Comment:
The command as is updates the timestamp of *all* messages in the table (there are no `WHERE ...`) limitations. You can also use `sqlite3` internal functions to set the timestamp, avoiding the call to `date`:

```suggestion
      sudo pihole-FTL sqlite3 "${FTLDBfile}" "UPDATE message SET timestamp = (cast(strftime('%s', 'now') as int)) WHERE message = '$1' AND type = 'GRAVITY';"
```

Reply:
'type = 'SCRIPT, see AdminLTE PR 2564. The goal is to make this possible from any script, including personal scripts. That is why the scriptname is included in the final message ($0). Can agree to change $(date +%s) into the suggestion, will change that, but can't signoff due to the type mismatch.